pkg/daemon/metrics.go

```diff
@@ -0,0 +1,31 @@
+package daemon
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var DefaultBindAddress = ":8080"
+
+func StartMetricsListener(addr string, stopCh chan struct{}) {
+	if addr == "" {
+		addr = DefaultBindAddress
+	}
+
+	glog.Infof("Starting metrics listener on %s", addr)
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", promhttp.Handler())
+	s := http.Server{Addr: addr, Handler: mux}
+	go func() {
+		if err := s.ListenAndServe(); err != nil {
+			glog.Exitf("Unable to start metrics listener: %v", err)
```

Comment:
This will abruptly `os.Exit()` the whole process, killing any pending `defer` in any other part of the application. I'm not sure if this this the expected behavior (versus more traditional error bubbling).

Reply:
thanks will double check and update.

Reply:
so i think maybe i should make an error chan send it to this func then send this message up as an error (not an exitf()) and check for it in start.go? not totally sure but seems to be the correct way to pass errors up from go routines?